rewriter/Visibility.cc

```diff
@@ -0,0 +1,46 @@
+#include "rewriter/Visibility.h"
+#include "ast/Helpers.h"
+#include "ast/ast.h"
+#include "core/Context.h"
+#include "core/Names.h"
+#include "core/core.h"
+#include "core/errors/rewriter.h"
+#include "rewriter/rewriter.h"
+
+using namespace std;
+
+namespace sorbet::rewriter {
+
+void Visibility::run(core::MutableContext ctx, ast::ClassDef *classDef) {
+    core::NameRef currentVisibility = core::Names::public_();
+
+    for (auto &stat : classDef->rhs) {
+        typecase(
+            stat.get(), [&](ast::Assign *assign) {},
+            [&](ast::Send *send) {
+                if (send->args.size() != 0 && send->recv != nullptr) {
+                    return;
+                }
+                // Technically this node can now be removed, but it doesn't really need to be removed now
+                if (send->fun == core::Names::private_()) {
+                    currentVisibility = core::Names::private_();
+                } else if (send->fun == core::Names::protected_()) {
+                    currentVisibility = core::Names::protected_();
+                } else if (send->fun == core::Names::public_()) {
+                    currentVisibility = core::Names::public_();
+                }
+            },
+            [&](ast::MethodDef *mdef) {
+                if (currentVisibility == core::Names::private_()) {
+                    ast::MethodDef::ARGS_store newArgs;
+                    newArgs.reserve(0);
+                    auto privateCall = ast::MK::Send1(mdef->loc, ast::MK::Self(mdef->loc), core::Names::private_(),
+                                                      ast::MK::Symbol(mdef->loc, mdef->name));
+                    classDef->rhs.insert(classDef->rhs.end(), std::move(privateCall));
```

Comment:
You are iterating over and modifying `classDef->rhs` in the same loop. Can we queue up the edits and process them after the loop?

Reply:
Ah okay -- yeah, Bo and I were wondering about this. @bss I can try to do this, but I might need some help with the syntax on this. If you are able to push to my branch or suggest edits that would work too.

Reply:
The syntax is just make yourself a `vector<TreePtr>` before your loop, collect all your newly minted send nodes into that vector with `.emplace_back(privateCall)` and then at the end of the loop, insert all the new private calls into the end of `rhs`.